lib/rules/web/ocm_console/cluster_detail_machinepool_tab.xyaml

```diff
@@ -121,4 +123,97 @@ click_scale_machine_pool_button:
     element:
         selector:
             xpath: //li/button[text()='Scale']
-        op: click
\ No newline at end of file
+        op: click
+check_edit_taints_in_dropdown:
+    action: expand_actions_of_machine_pool
+    element:
+        selector:
+            xpath: //button[text()='Edit taints']
+click_edit_taints_in_dropdown:
+    element:
+        selector:
+            xpath: //button[text()='Edit taints']
+        op: click
+check_no_default_in_dropdown_when_editing_taints:
+    elements:
+        - selector:
+            xpath: //select[@id='machinePoolId']
+          op: click
+        - selector:
+            xpath: //option[text()='Default']
+          missing: true
+check_for_taints_row:
+    elements:
+        - selector:
+            xpath: //input[@id='taints[<row_num>].key']
+        - selector:
+            xpath: //input[@id='taints[<row_num>].value']
+        - selector:
+            xpath: //select[@id='taints[<row_num>].effect']
+check_edit_taints_dialog:
+    action: click_edit_taints_in_dropdown
+    action: check_no_default_in_dropdown_when_editing_taints
+    action: check_for_taints_row
+    elements:
+        - selector:
+            xpath: //h4[text()='Your edits will only apply to new nodes']
+        - selector:
+            xpath: //span[text()='Edit taints']
+        - selector:
+            xpath: //button[text()='Add taint']
+        - selector:
+            xpath: //button[text()='Save' and @aria-disabled='true']
+        - selector:
+            xpath: //button[text()='Cancel']
+click_and_check_add_taint:
+    elements:
+        - selector:
+            xpath: //button[text()='Add taint']
+          op: click
+        - selector:
+            xpath: (//button[@class='pf-c-button pf-m-link reduxFormKeyValueList-removeBtn'])[2]
+    action: check_for_taints_row
+click_delete_taint_row_button:
```

Comment:
We have defined click_delete_taint_button  in file lib/rules/web/ocm_console/dialogs_and_sidebar.xyaml